java/src/main/java/io/cucumber/gherkin/GherkinTokenMatcher.java

```diff
@@ -46,25 +46,21 @@ public void reset() {
         currentDialect = dialectProvider.getDefaultDialect();
     }
 
-    private GherkinDialect getCurrentDialect() {
-        return currentDialect;
-    }
-
-    private void setTokenMatched(Token token, TokenType matchedType, String text, String keyword, Integer indent, StepKeywordType keywordType, List<GherkinLineSpan> items) {
+    private void setTokenMatched(Token token, TokenType matchedType, String text, String keyword, int indent, StepKeywordType keywordType, List<GherkinLineSpan> items) {
         token.matchedType = matchedType;
         token.matchedKeyword = keyword;
         token.keywordType = keywordType;
         token.matchedText = text;
         token.matchedItems = items;
-        token.matchedGherkinDialect = getCurrentDialect();
-        token.matchedIndent = indent != null ? indent : (token.isEOF() ? 0 : token.line.getIndent());
```

Comment:
I'm assuming you used an automated refactoring tool to do this? If so, ok. If not, please revert and redo with a tool.

Reply:
This is hand-made refactoring without a tool. I understand that you trust a tool more than a developer for such refactoring, but here the replacement is quite simple: when the `indent` argument is passed with `null` value, replace by `token.line.getIndent()` (except in `match_EOF(Token)` where we replace by 0).

I agree that tool-based refactoring are useful, typically for large-scale refactoring. But here it was really simple, and I don't know which refactoring tool to use.

Reply:
I was suspicious because this method has 10 calls sites, and only 8 were updated. Now I've done the refactoring with IDEA (Eclipse would work too) and it looks all correct. The two remaining call sites match `Other` and `Comment` tokens but those use a constant indent of 0 

The recipes I used

1. Extract `indent != null ? indent : (token.isEOF() ? 0 : token.line.getIndent());` to a local variable at the top of the `setTokenMatched` method.
2. Extract `setTokenMatched2` from `setTokenMatched` excluding that new line. 
3. Inline `indent != null ? indent : (token.isEOF() ? 0 : token.line.getIndent());` again.
4. Inline `setTokenMatched`
5. Rename `setTokenMatched2` to `setTokenMatched`
6. Resolve all constant conditions at `setTokenMatched` invocations.